File: doc-examples/nitrite-example-java/src/main/java/example/VersionedBook.java
package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.Version;

// tag::versioned-book[]
@MappedEntity
public class VersionedBook {
    @Id
    @GeneratedValue
    private String id;

    private String title;

    @Version
    private Long version; // <1>

    public VersionedBook() {
    }

    public VersionedBook(String title) {
        this.title = title;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Long getVersion() {
        return version;
    }

    public void setVersion(Long version) {
        this.version = version;
    }
}
// end::versioned-book[]

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/VersionedBook.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.Version

// tag::versioned-book[]
@MappedEntity
class VersionedBook {
    @Id
    @GeneratedValue
    var id: String? = null

    var title: String = ""

    @Version
    var version: Long? = null // <1>

    constructor()

    constructor(title: String) {
        this.title = title
    }
}
// end::versioned-book[]
